src/sentry/sentry_metrics/querying/api.py

```diff
@@ -144,9 +145,36 @@ def meta(self) -> Sequence[Mapping[str, str]]:
         return self.result["meta"]
 
 
-class QueryParser:
+class MutableTimeseries:
     PERCENTILE_REGEX = re.compile(r"^p(\d{1,3})$")
 
+    def __init__(self, timeseries: Timeseries):
+        self._timeseries = timeseries
+
+    def inject_environments(self, environments: Sequence[Environment]) -> "MutableTimeseries":
+        if environments:
+            environment_names = [environment.name for environment in environments]
+            existing_filters = self._timeseries.filters[:] if self._timeseries.filters else []
+            self._timeseries = self._timeseries.set_filters(
+                existing_filters + [Condition(Column("environment"), Op.IN, environment_names)]
+            )
+
+        return self
+
+    def alias_operators(self) -> "MutableTimeseries":
+        # In case we have a percentile in the form `px` where `x` is in the range [0-100], we want to convert it to
+        # the quantiles operation which generalizes any percentile.
```

Comment:
That is incorrect. We support a only pre-defined set of percentiles, hence also the hard-coded functions in the public API. There's no generic `pXX` function conversion. If we want generic percentiles, then let's expose a `percentile(X, <metric>)` function, instead. Likewise, there is no `p100` function. There is just `max`.

You could consider adding a hard-coded mapping of the four percentile functions, which also gets rid of this regex.

Reply:
We wanted to have a discussion about which percentiles we want to expose and how but the idea would be that the layer itself performs these coercions.